mev_inspect/inspect_block.py

```diff
@@ -53,13 +53,15 @@
 from mev_inspect.schemas.transfers import Transfer
 from mev_inspect.swaps import get_swaps
 from mev_inspect.transfers import get_transfers
+from mev_inspect.utils import RPCType
 
 logger = logging.getLogger(__name__)
 
 
 async def inspect_block(
     inspect_db_session: orm.Session,
     w3: Web3,
+    type: RPCType,
```

Comment:
rather than passing the rpc type as an argument, is it possible to retrieve a list of added middlewares from the w3 object itself, and determine the type from the presence/absence of the geth_poa_middleware?

Reply:
I agree, will commit new changes